Source/Controllers/DataExport/SPExportController.m

```diff
@@ -3162,27 +3117,11 @@ - (BOOL)applySettingsFromDictionary:(NSDictionary *)dict error:(NSError **)err
 	// look up that bookmark and request access
 	if(bookmarks.count > 0){
 		if((o = [dict objectForKey:@"exportPath"])) [exportPathField setStringValue:o];
-		
-		NSError __block *error = nil;
-		
-		[self.bookmarks enumerateObjectsUsingBlock:^(NSDictionary *dict2, NSUInteger idx, BOOL *stop) {
-			
-			NSString *tmpStr = [NSURL fileURLWithPath:[exportPathField stringValue] isDirectory:YES].absoluteString;
-			
-			if(dict2[tmpStr] != nil){
-				self.userChosenDirectory = [NSURL URLByResolvingBookmarkData:dict2[tmpStr]
-																	 options:NSURLBookmarkResolutionWithSecurityScope
-															   relativeToURL:nil
-														 bookmarkDataIsStale:nil
-																	   error:&error];
-				*stop = YES;
-			}
-		}];
-		
-		// if no bookmark was found this just calls against nil
-		if(!error){
-			[userChosenDirectory startAccessingSecurityScopedResource];
-		}
+
+        NSString *fileURLString = [NSURL fileURLWithPath:[exportPathField stringValue] isDirectory:YES].absoluteString;
+
+        // ret value can be nil
+        userChosenDirectory = [SecureBookmarkManager.sharedInstance bookMarkForFilename:fileURLString];
```

Comment:
A little pedantic, but "bookmark" is one word not "book mark" so the camel-cased function arg label should probably be `bookmarkForFilename`

Reply:
sorted. Also `bookMarkForURL`